test/lightning/usage_tracking/report_data_test.exs

```diff
@@ -203,84 +203,62 @@ defmodule Lightning.UsageTracking.ReportDataTest do
   defp build_hash(uuid), do: Base.encode16(:crypto.hash(:sha256, uuid))
 
   defp build_project(count) do
-    project =
-      insert(
-        :project,
-        project_users: (&build_project_user/1) |> repetitions(count)
-      )
+    project = insert(:project, project_users: build_project_users(count))
 
-    fn _ignore -> insert_workflow(project, count) end |> repetitions(count)
+    workflows = insert_list(count, :workflow, project: project)
 
-    project
-  end
-
-  defp repetitions(builder, repetitions) do
-    1..repetitions |> Enum.map(builder)
-  end
-
-  defp build_project_user(_ignore) do
-    build(:project_user, user: build(:user))
-  end
-
-  defp insert_workflow(project, count) do
-    workflow =
-      insert(
-        :workflow,
-        project: project,
-        jobs: (&build_job/1) |> repetitions(count)
-      )
+    for workflow <- workflows do
+      [job | _] = insert_list(count, :job, workflow: workflow)
+      work_orders = insert_list(count, :workorder, workflow: workflow)
 
-    # Pass around a job so that we do not have a bottom-up creation of workflows
-    # from the step factory. Currently, jobs are not pat of the path towards
-    # finding runs so any job will do
-    [job | _] = workflow.jobs
+      for work_order <- work_orders do
+        runs =
+          insert_runs(
+            count: count,
+            project: project,
+            work_order: work_order,
+            job: job
+          )
 
-    fn _ignore ->
-      insert_work_order(project, workflow, job, count)
+        for run <- runs, do: insert_steps(count, run, job)
```

Comment:
If doesn't take much time or next time you can insert the run along with the steps like this:
https://github.com/OpenFn/lightning/blob/9fbb62c2e1a9c85feee153058f1b418392ed2072/test/lightning_web/live/work_order_live_test.exs#L59

Reply:
@jyeshe Done!